lib/experimental/class-wp-navigation-fallbacks-gutenberg.php

```diff
@@ -0,0 +1,241 @@
+
+		if ( $fallback && ! is_wp_error( $fallback ) ) {
+			return $fallback instanceof WP_Post ? $fallback : static::get_most_recently_published_navigation();
+		}
+
+		$fallback = static::create_default_fallback();
+
+		if ( $fallback && ! is_wp_error( $fallback ) ) {
+			return $fallback instanceof WP_Post ? $fallback : static::get_most_recently_published_navigation();
+		}
+
+		return null;
+	}
+
+	/**
+	 * Finds the most recently published `wp_navigation` Post.
+	 *
+	 * @return WP_Post|null the first non-empty Navigation or null.
+	 */
+	private static function get_most_recently_published_navigation() {
+
+		// Default to the most recently created menu.
+		$parsed_args = array(
+			'post_type'              => 'wp_navigation',
+			'no_found_rows'          => true,
+			'update_post_meta_cache' => false,
+			'update_post_term_cache' => false,
+			'order'                  => 'DESC',
+			'orderby'                => 'date',
+			'post_status'            => 'publish',
+			'posts_per_page'         => 1, // get only the most recent.
+		);
+
+		$navigation_post = new WP_Query( $parsed_args );
+
+		if ( count( $navigation_post->posts ) > 0 ) {
+			return $navigation_post->posts[0];
+		}
+
+		return null;
+	}
+
+	/**
+	 * Creates a Navigation Menu post from a Classic Menu.
+	 *
+	 * @return int|WP_Error The post ID of the default fallback menu or a WP_Error object.
+	 */
+	private static function create_classic_menu_fallback() {
+		// See if we have a classic menu.
+		$classic_nav_menu = static::get_fallback_classic_menu();
+
+		if ( ! $classic_nav_menu ) {
+			return new WP_Error( 'no_classic_menus', __( 'No Classic Menus found.', 'gutenberg' ) );
+		}
+
+		// Todo: inject as dependency.
+		$menu_converter = new WP_Nav_Menu_Conversion_Gutenberg( $classic_nav_menu );
+
+		// If we have a classic menu then convert it to blocks.
+		$classic_nav_menu_blocks = $menu_converter->convert();
+
+		if ( empty( $classic_nav_menu_blocks ) ) {
+			return new WP_Error( 'cannot_convert_classic_menu', __( 'Unable to convert Classic Menu to blocks.', 'gutenberg' ) );
+		}
+
+		// Create a new navigation menu from the classic menu.
+		$wp_insert_post_result = wp_insert_post(
+			array(
+				'post_content' => $classic_nav_menu_blocks,
+				'post_title'   => $classic_nav_menu->name,
+				'post_name'    => $classic_nav_menu->slug,
+				'post_status'  => 'publish',
+				'post_type'    => 'wp_navigation',
+			),
+			true // So that we can check whether the result is an error.
+		);
+
+		return $wp_insert_post_result;
+	}
+
+	/**
+	 * Determine the most appropriate classic navigation menu to use as a fallback.
+	 *
+	 * @return WP_Term|null The most appropriate classic navigation menu to use as a fallback.
+	 */
+	private static function get_fallback_classic_menu() {
+		$classic_nav_menus = wp_get_nav_menus();
+
+		if ( ! $classic_nav_menus || is_wp_error( $classic_nav_menus ) ) {
+			return null;
+		}
+
+		// 1.
+		// Attempt to use the menu assigned to location `primary`.
```

Comment:
Since we created these small functions precisely to give them specific self documenting names the comments are not helping, on the contrary they get in the way.

Reply:
Haha yes. Since I extract methods the comments are now redundant 🎉

Reply:
✅